pkg/promtail/targets/objectstore/object_reader.go

```diff
@@ -0,0 +1,245 @@
+package objectstore
```

Comment:
Will this support other object stores in the future or should it be named `s3`?

Reply:
@jeschkies Yes